server/src/main/java/io/crate/planner/optimizer/rule/EquiJoinToLookupJoin.java

```diff
@@ -179,12 +177,12 @@ private static LogicalPlan createLookup(Collect smallerRelation,
             List.of(largerRelationColumn, lookUpQuery),
             DataTypes.BOOLEAN
         );
-        WhereClause whereClause = largerRelation.where().add(anyEqFunction);
-        LogicalPlan largerSideWithLookup = new Collect(largerRelation.relation(), largerRelation.outputs(), whereClause);
-        LogicalPlan smallerSideIdLookup = new Collect(smallerRelation.relation(), List.of(smallerRelationColumn), smallerRelation.where());
-        smallerSideIdLookup = new RootRelationBoundary(smallerSideIdLookup);
-
-        return MultiPhase.createIfNeeded(Map.of(smallerSideIdLookup, lookUpQuery), largerSideWithLookup);
+        WhereClause whereClause = largerSide.where().add(anyEqFunction);
+        LogicalPlan largerSideWithLookup = new Collect(largerSide.relation(), largerSide.outputs(), whereClause);
+        var smallerSideIdLookup = new RootRelationBoundary(smallerSide);
```

Comment:
Couldn't we also already remove the `Collect` restriction for the larger side by wrapping it in a `Filter` operator instead of extending the `where()` ?